Fee center schema and API

DedrickEnc opened the pull request:
This PR adds an initial schema and API to continue the feature developing
Here are things to notice in this PR : 

- The table ```cost_center``` and ```profit_center``` are deleted
-  A new table ```fee_center``` is created```
- The modules cost center and profit center are deleted
- The service module is updated to fit with the  changes
- The delete functionality is not yet implemented and the integration  test for this functionality is ignored, because we did not talk about that during the design. 
Thank you for contributing!


Before submitting this pull request, please verify that you have:
 - [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

sfount committed:
(chore) Use constants for filter bar expand/ collapse

This commit replaces hardcoded string style values in the posting
journal util filter bar collapse/ expanded states.

sfount committed:
(Chore) Limit to 1 edit; add on focus select

sfount committed:
Add basic edit utility bar

This commit introduces the concept of an edditing utility bar.
It adds edit and save methods into the controller to connect
resizing the grid for the utility bar and calling the edit and save
methods on the service.

sfount committed:
Implement aggregate data flag for journal transactions

This commit adds the optional flag `aggregates` to the `/journal`
route. Setting the flag `aggregates=1` will now return a javascript
object with `journal` (rows) and `aggregate` data. Aggregate data will
provide the sum of debits and credits as well as the total number of
rows for every transaction in the journal query.

Aggregates have been tested to work for both standard requests and with
any of the filters applied.

sfount committed:
Update client side journal to request aggregates

This commit extends the Journal Service `read` method to ensure
aggregates=1 is always passed the server for the journal module.
This updated method is named `grid`.

sfount committed:
Update transaction debits and credits; aggregates + header row

This commit introduces a basic pre processor for posting journal data,
this allows the header rows to show transaction level data even when
the journal is being filtered.

It also updates to using `uiGridGroupingConstants` aggregation types,
these aggregation methods are not altered when rows are expanded/
collapsed in the grid. This partially addresses #1041.

Each row now tracks information about the transaction as a whole.
This is used to include formation about the transaction in the row
header and also addresses #1041.

Closes #1041.
Closes #1045.

sfount committed:
Register column and row processors to assign aggregates/ edit styles

This commit ensures that aggregates are assigned using
`uiGridGroupingConstants`, it also configures these using processors
so that they are run when required.

Refactor indexing and updating editing styles to ensure that a
transaction being edited will always be displayed correctly.

DedrickEnc committed:
restructuring the data base

sfount committed:
Adding rows to a transaction currently being edited

This commit exposes the add a row method on the Transaction
service to the main controller. This allows the currently edited
transaction to insert additional rows. Shared information required
for additional rows is cached on the entity being edited, it can
then be used to drive new rows.

DedrickEnc committed:
updating test

DedrickEnc committed:
fixing integration test

DedrickEnc committed:
interface implementation done

DedrickEnc committed:
update service module and test

mbayopanda committed:
Purchase order registry

mbayopanda committed:
Implement search on purchase registry

mbayopanda committed:
Modal for search in purchase order registry

mbayopanda committed:
search filter implementation

mbayopanda committed:
Service for formating filter data

mbayopanda committed:
Fix search filter format

mbayopanda committed:
Remove temporary print pdf of registry

mbayopanda committed:
Fix errors

mbayopanda committed:
Fix and enhancement of purchase order registry

sfount committed:
feat(journal) Downloading records for transaction edit

sfount committed:
feat(journal) Deleting rows; server skeleton

This commit implementes the basic infrastructure for writing edits to
the database on the server. It also updates the client to resolve issues
with deleting and filtering during editing.

sfount committed:
feat(journal) Map human readable edit values -> id

This commit proposes a way to allow users to edit human readable values
on the posting journal client and have the server convert these into
dateabase ready entity uuids.

sfount committed:
feat(journal) Map reference and entity ids

This commit updates the editing method on the server to map all columns
that can be edited with human readable formats (requiring mapping to
ids). It also includes edits to journal rows in the mapping process so
that both edits and new rows are treated in the same way.

Commit:
fix(vouchers): print correct currency & user

This commit fixes the voucher registry so that it prints the correct
currency in the printed report.  It also ensures that the report prints
out the username in the user column.

Fixes #187.

lomamech committed:
Link between registry of patients to Cash Registry
Update cash.routes adding property filter and display
Update payments/registry with the using of $state.params`

lomamech committed:
Fix property fix clearFilters for cash registry

lomamech committed:
Link between Registry of patient to Patient Invoice
- Add property debtor uuid to PatientInvoicesService in format filter
  parameters
- Update patient invoice registry to filterData with Parameter of link who
  come from Patient Registry
- Update seach modal for research of Invoices by the name of Patient
- Update app.js for the controller patientInvoices adding the property
  params

mbayopanda committed:
e2e tests for purchases registry

mbayopanda committed:
fix lint error

mbayopanda committed:
fix e2e test

lomamech committed:
Set $tate.params null instead {}

sfount committed:
feat(journal) Cancel transaction editing

This commit adds logic for cancelling transactions. Edit/ change objects
are reset in the Transaction service and styles are updated to reflect
the status. The main Journal controller querries the server to ensure it
is up to date and valid for the current filter.

lomamech committed:
Sanitaze code
Using FILTER_BAR_HEIGHT
Remove the es6 from client

lomamech committed:
Fix error

lomamech committed:
Include Cash Payment and Voucher Identifiers in Patient Activity report
- Update api debtors index, make jointure with table cash and voucher
- Update financial patient handlebars with the new property document
- Using alternative structure for set the value for document

DedrickEnc committed:
integrating suggestion

Commit:
feat(db): administrative utilities

This commit creates administrative utilities for the database that are
useful in correcting database issues.  It includes the ability to repost
every single transaction type as well as recalculate the entity maps.

lomamech committed:
Using document_map get the DocumentId

Commit:
feat(cash): add created_at timestamp

This commit adds the created_at timestamp to the cash table.

Closes #1137.

lomamech committed:
Remove the ui-select for modal filter search
Remove the link from DebtorService

DedrickEnc committed:
updating contnues

Commit:
fix(vouchers): bug/UX fixes on findReference Modal

This commit fixes usability issues on the findReference modal.
  1. Makes the header an `<ol>` headercrumb
  2. Uses the `translate` directive
  3. Adds a loading indicator to the grid.
  4. Makes sure the reference identifier is clearly visible in grid.

Closes #1127. Closes 1139.

lomamech committed:
Delete the cash registry test by Client

DedrickEnc committed:
many minors fixes

DedrickEnc committed:
commenting code to pass jshint for this PR

jniles committed:
Update Install.md

This commit updates INSTALL.md with further testing instruction and brings it up to date with the needed dependencies.

lomamech committed:
Using format dd/mm/yyyy
Update app.js putting property yearOptions and dayOptions
Update component bhDateEditor and bhDateInterval
Update some element to use bhConstants for the using of dayOptions.format

jniles commented on server/controllers/finance/feeCenter.js:
```diff
@@ -0,0 +1,251 @@
+    if(filter !== ''){
+      filter += ' AND ';
+    }
+    filter += `fc.is_cost = ${req.query.is_cost} `;
```
I have pulled your branch typed this into my web browser.
```
http://localhost:8080/fee_centers?is_cost=1 UNION SELECT id, password as label, active AS is_cost FROM user AS fc ; --
```
And this is the result I got:
![sql injection attack](https://cloud.githubusercontent.com/assets/896472/22788931/b6495722-eee1-11e6-8233-4fdab507dfae.png)
_Fig 1: An SQL Injection Attack_

These attacks are quite common across the web.  We have a library called `db` that will parse the parameters and prevent sensitive information like passwords from leaking.

DedrickEnc replied:
I have written an email

Commit:
fix(cashbox): ensure is_auxiliary is updated

This commit should fix the issues in cashbox management where cashboxes
were not properly updated with the is_auxiliary field.  It also
generally improves the code by:
 1. Using the translate directive on the client.
 2. Using `db.one()` on the server.
 3. Using `FilterParser` on the server.

Closes #976.

Commit:
refactor(cash): update transfer modal to standards

This commit guts and rewrites the cash transfer modal for better error
handling and to add a few new features.

**Notable Changes**
 1. The transfer modal is now powered by the Voucher Form.  This gives
 several advantages:
  1. Better error messages and handling - all of these are gotten for
  free from the VoucherForm service.
  2. A better description - a new template using $translate was
  developed based on feedback from @imawh.
 2. `translate` directive replaces the `translate` filter
 3. Fixed updating bug with currency_id where model lagged behind the
 `bhCurrencySelect` by one increment.
 4. Removed transfer code from `CashService`.
 5. Fixed "TO" and "FROM" label text bug.
 6. Print a JV receipt when the transfer is complete.

Closes #872.

lomamech committed:
Improve some E2E test and fix patient registry modal

lomamech committed:
Using icon fa-filter instead fa-search
Using btn-info when the filtering is activate

Commit:
refactor(db): remove entity_type from gl+pj

This commit removes the column `entity_type` from the general ledger and
the posting journal.  In order to do this, many of the tests for trial
balance failure are no longer relevant.  I have either removed or
skipped them until we can write stricter tests.  Since the posting
journal is being actively worked on, this seems appropriate.

Partially addresses #584.

Commit:
refactor(uiGrid): registry dropdowns anchors

This commit refactors the ui-grids in the application to use a[href]
links for their dropdowns. Hyperlinks are more familiar to users than
triangles.  The affected grids are users, inventory list, and Billing
Services.

It also limits the height of the inventory modal to a more sensible grid
height.

Commit:
refactor(uiGrid): registry dropdowns anchors

This commit refactors the ui-grids in the application to use a[href]
links for their dropdowns. Hyperlinks are more familiar to users than
triangles.  The affected grids are users, inventory list, and Billing
Services.

It also limits the height of the inventory modal to a more sensible grid
height.

Commit:
fix(db): build TEMP tables before posting

This commit fixes a bug specifically in the administrative tools that
builds temporary tables required for posting cash payments before
the zRepostCash() method attempts to repost them.  This also helps
modularize the procedure code.

Commit:
deps(eslint): rm jshint; install eslint

This commit removes our dependency on jshint and replaces it with eslint
as a linter.  Eslint is much stricter and will reduce the work of a code
reviewer in commenting on style.  The current codebase will need to be
migrated significantly and rulesets customized for this to be an
effective tool.

Closes #223.

Commit:
refactor(eslint): rm 'use strict'

The command was this:
```sh
grep -rl "'use strict'" | xargs sed -i /'use strict'/d
```

Since we are now using the `use-strict` module, we will no longer need
to put use strict everywhere.

Commit:
refactor(eslint): make eslint failures noncritical

This commit temporarily turns off eslint failures while the code base
can be migrated.  ESLint failures will be reported in Travis, but will
not break the build.

Commit:
docs: rm mentions of jshint

Since we are now using ESLint, might as well clear out jshint mentions.

Also runs the command:
```sh
grep -rl "jshint" | xargs sed -i /jshint/d
```

... on select folders.

Commit:
refactor(balance): use translate directive

This commit refactors the balance report modal to use the `translate`
directive and proper HTML the way bootstrap prescribes.

Commit:
feat(reports): Open Debtors Report

This commit implements the server and client sides to the open debtors
report.  It provides a list of all debtors who have unpaid debts to the
enterprise.

The current implementation lacks proper translations as well as proper
SQL filtering for unposted records.  To complete this, there needs to be
 - [ ] i18n
 - [ ] Integration Tests
 - [ ] End to End Tests
 - [ ] General Report Refactoring (Notify should use report keys)

Closes ....

lomamech committed:
Optimisation with the using of table

lomamech committed:
Optimisation of SQL query

Commit:
fix(filters): ensure dates are parsed

This commit fixes a bug in the filter parser library such that it
converts dates for the client into new Date() objects.

Commit:
fix(filters): ensure dates are parsed

This commit fixes a bug in the filter parser library such that it
converts dates for the client into new Date() objects.

Commit:
refactor(reports): i18n + linting open debtors

This commit lints the open debtors report and adds translation keys.
Integration tests are added to future-proof the report.  Finally, the
report is added to the database records.

mbayopanda committed:
fix mergin conflicts

mbayopanda committed:
fix and enhancements

lomamech committed:
Improve inventory groups
Update Inventory group service for get option include_members
Sanitaze configuration groups.js removing unused function

lomamech committed:
Update sql request for getGroupsMembers

lomamech committed:
Fix invalid HTML and using of class text-lowercase

Commit:
fix(balance): render report in enterprise currency

This commit updates the balance report to use the enterprise currency
instead of the default currency.

Closes #1168.

Commit:
fix(vouchers): ensure amount is clear in simple

This commit fixes a bug where the simple voucher amount was not cleared
on submit.

Commit:
fix(vouchers): render the amount as currency

This commit ensures that the voucher registry renders the amounts as
currencies.

Partially addresses #671.

lomamech committed:
Login error messages (#1175)

* Login error messages
Update auth.js to catch No Permissions and No Project
Add two lib errors for No Permission and No Project

* Update test integration for recognized user without a project

* Resolve conflict in Unauthorized

* Using db exec instead transaction

Commit:
refactor(purchases): implement filter parsing

This commit leverages FilterParser for the purchase registry.  It also
implement integration tests for the `/search` route.  Finally, it
ensures that purchase orders abides by the new identifiers syntax.

Supersedes #1114.  Closes ##1114.

Commit:
fix(voucher): prise en charge reference bug

This commit fixes a bug in the prise en charge module due to naming
confusion of columns in the voucher table.

mbayopanda committed:
fix currency of cashflow report

Commit:
Merge branch 'master' into feature-purchase-order-registry

Commit:
fix(report): aged debtor maths

This commit fixes the aged debtor maths by ensuring that dates are
properly compared and ranges are restricted to avoid double-counting.

Closes #1169.

lomamech committed:
Fix Display in Invoices
- Update the request in PatientInvoice invoiceBillingQuery adding total of invoice
-Update receip.handlebars
`

Commit:
fix(purchases): use proper checkbox html

This commit fixes the purchase registry to use bootstrap-compliant HTML in
its search modal.

Commit:
feat(cash): filter out reversed records

This commit uses the FilterParser to filter out reversed records on the
cash payments page.  It also gives the option to exclude all records but
those that are reversed.

Closes https://github.com/Vanga-Hospital/bhima-2.X/issues/19.

Commit:
refactor(creditNote): nice credit note text

This commit provides nicer credit note text when reversing records.  It
also removes the original invoice description from the credit note
description as this is redundant and it's usefulness has not come up in
practice.

Closes https://github.com/Vanga-Hospital/bhima-2.X/issues/249.

Commit:
fix(vouchers): use document instead of reference

The complex voucher page was incorrectly updated to use references when
the backend tables use 'document' as the column name.  This caused a lot
of confusion and a bug in production.  This commit goes through and updates
all clientside references to use "document" explicitly.

Closes https://github.com/IMA-WorldHealth/bhima-2.X/issues/1194.

lomamech committed:
Update receipt A4 for cash paiement, invoices and voucher, Add proper… (#1201)

* Update receipt A4 for cash paiement, invoices and voucher, Add property Barcode

* Remove duplicate <head>

* Using head from head.handlebars

Commit:
deps(angular): upgrade to 1.6.x

This commit migrates the entire application to angular 1.6.x.  The most
important change is that all components must use component lifecycle
hooks such as $onInit, $onDestroy, etc.

This commit also upgrades ui-router to v1.x.x.  This makes sure
ui-router does not yield `unhandledrejection` errors.  The changes are
mostly hidden at the moment.

Finally we've enabled some performance improvements:
 1. Turn off the parsing of comment directives
 2. Turn off the parsing of class directives
 3. Use $onDestroy() lifecycle hooks to free up component events

Closes #1053. Closes #1086. Closes #1051.

mbayopanda committed:
fix account report (#1203)

* fix account report

* eslint fixing

* fix sum calculations

* enhancement of account report

Commit:
fix(test): update tests to standards + ng 1.6.x

This commit upgrades our tests to use ES6 for many of the tests, plus
use the new hashbang routing of angular 1.6.x.  It also includes the
stateEvents.js file while we migrate to the new ui-router transitions
onEnter and onExit states.

One failing test to go!

Commit:
fix(trialbalance): transition out of TB on post

This commit resolves conflicting $state transitions by eliminating the
transition on failure.  This means that when $uibModalStack.dismissAll()
takes affect, the $state is not triggered to transition.  This is what
you expect.

Commit:
fix(test): skip failure bhBreadcrumb Karma tests

This commit skips failing karma tests until they can be investigated.

DedrickEnc committed:
refactor continues

Commit:
refactor(cashbox): use bhAccountSelect component

This commit refactors the cashbox page to use the bhAccountSelect
component.  The end to end tests have been updated accordingly.

Commit:
refactor(creditors): bhAccountSelect for groups

This commit refactors the creditor groups page to use the
`bhAccountSelect` component.

Commit:
refactor(enterprises): use bhAccountSelect

This commit migrates the Enterprise Module to using bhAccountSelect.

Commit:
refactor(subsides): use bhAccountSelect

This commit updates subsidies to using `bhAccountSelect` for their
account selection.

Commit:
Merge branch 'master' into feature-purchase-order-registry

Commit:
fix(purchase): update test references

This commit updates the test references to the new standard.

jniles committed:
Merge pull request #1186 from jniles/feature-purchase-order-registry

feature: purchase order registry

Commit:
fix(cash): bug in debtor $state transitions

This commit fixes another bug caught in $state transitions due to modals
being incorrectly $transitioned out of.  This makes sure the end to end
tests work correctly.

Commit:
feat(env): turn off events

This commit adds environmental variable control of the events emitted by
Topic.  By setting the variable `DISABLE_EVENTS` to false, no Redis
subscriptions will be made at runtime.  In fact, the code is read as
dead code and returns immediately on entry into the function.

Closes #948.

Commit:
feat(build): build with revision control

This commit builds the gulp file with revision control such that our
caching scheme can be implemented at infinite and reduce network load.

Closes #1219.

lomamech committed:
Fix patient links from cash and Invoice Registry

sfount committed:
refactor(links) Link parent modules in headercrumb

This commit adds anchor tags to the headercrumb*(tm)* module title for
the following modules:
* Patient editing -> Patient Registry
* Patient records -> Patient Registry
* Debtor Group Management Update -> Debtor Group Managment Lists

It also updates the `<a>` style for the headercrumb to differentiate a
link. This style can be refined in the future.

This commit also fixes a number of links that were broken with
dependency updates.

sfount committed:
refactor(patients) Registry link records/cards

This commit removes the double click to open the patient card on the
patient registry. To maintain the same functionality the card is now
linked from the patient reference.

It also refactors the dropdown actions menu to include the reference and
use the new identity card icon.

Commit:
refactor(trial balance): overall improvements

This commit improves the Trial Balance UI and UX in the following ways:
 1. The submit button now disables and enters a loading state after
 posting to the GL.
 2. The grid now right aligns the currency amounts.
 3. The grid now uses currency rendering for all amounts.
 4. The buttons now behave responsively.
 5. The labels no longer bleed into the headercrumb.

There are also some performance improvements:
 1. Use flat entity access on the grids
 2. Use fastWatch
 3. Use translate directive where possible

Addresses some of the points in #1163.  Closes #1171.

sfount committed:
feat(patients) Add date of birth field to records

This commit adds the date of birth field to the patient record module.
It also includes a disclalimer link if only the year of birth was
specified during patient registration.

Commit:
fix(trial balance): bug in totals calculation

This commit fixes a bug in the totals calculation for the trial balance
due to an incorrect LEFT JOIN.  This has been broken out into multiple
subqueries.  The performance of these queries should be investigated,
but there are no obvious optimizations to make.

sfount committed:
feat(receipts) Add service line to invoice receipt

This commit adds the service that the invoice was created for to both
the invoice POS and standard receipt. On the standard receipt it
replaces the duplicated amount field, this is so that there is one point
of truth for the total amount on the receipt, it can be undone if
required.

sfount committed:
fix(registry) Fix refersh issue on invoices/cash

This commit ensures that when an invoice or cash payment is reversed the
registry still filters appropriately.

It also resolves a bug where filters could not be removed when passed in
through a parameter link. This was resolved by standardising passing
parameters by loading them in a startup method and not complicating the
load/ search logic.

Cash payments previously used many different techniques for filtering
results. It has been standardised and now caches filters.

sfount committed:
fix(filters) wrap reverse filter in ()

This commit wraps the reverse filter (currently used on credit notes) in
parenthesis to ensure that it respects additional filters currently
applied. Currently the OR logic in the filter has an impact on the
string of AND conditions.

Commit:
refactor(trial balance): accountant feedback

This commit encorporates feedback from our accountant.
 1. Grid Footer is only shown when there are imbalances.
 2. Totals are only ever shown for debit/credit and not for
 before/after.
 3. The grid is no longer padded - slightly more screen real estate for
 users.
 4. The account now includes the account text.
 5. Removed Group By Transaction

sfount committed:
fix(debtors) remove pagination restriction/debtors

This commit removes paginiation from the Debtor Group maangement page.
It resolves the issue with the pagination bar not previously showing but
also comments out pagination functionality. Currently the pages and
pagination do not respect the filter - this can lead to a very confusing
user experience.

sfount committed:
fix(debtors) fix promise logic on group deletion

This commit fixes the order of promise logic when deleting a debtor
group. With this structure the Notification and refresh methods will
only be called if the Debtor Group delete method is called.

mbayopanda committed:
General ledger and Posted Journal (#1212)

* new general ledger

* Account Slip Fixing and Posted Journal

* Fix and enhancements

* fix eslint

* Fix and enhancement

* Fix parent of posted journal

sfount committed:
refactor(journal) update cases for row digest

Commit:
fix(bhRendererDropdown): rebuild URL on selection

This commit instructs the bhRendererDropdown to rebuild the URL
parameters when the users changes the dropdown menu selection.

Closes #1239.

sfount committed:
(chore) Re-write large object assignments as map

This commit removes large an unmaintainable sections of code, it also
ensures that digestion calculations are always correct for the current
edit session.

Commit:
feature(vouchers): implement FilterParsing

This commit implements the FilterParser on the voucher registry and also
updates the integration test to behave as expected.  This includes
searching for the voucher by the full reference and not simply the
reference number.

It also removes the getVouchers() code and replaces it with
lookupVoucher().  This is much cleaner and results in a nicer looking
receipt.

Finally, it refactors the voucher report to be a bit nicer to view/read.

lomamech committed:
Using fa-close instead fa-ban

lomamech committed:
Suggestion account report (#1211)

* - Adding specific date for the runnig balance
- Update strucuteof reportAccounts

* Resolve conflict Report Account index

* resolve conflict with translate file

* Update report Accounts

* Calculate cumulbalance from DBMS and not in javascript
Improve sql for getAccountTransactions

* Using bh-account-select

* Update test E2E repport accounts

mbayopanda committed:
Fix currency conversion and date aggrega (#1234)

* Fix currency conversion and date aggrega

* fix linting exceptions

* hide aggregation label in journal

* fix number in journal

Commit:
fix(accounts): fix rendering bugs

This commit fixes two bugs in the account report where cumulative
summation and the description were not rendering due to incorrect SQL.
It also ensures that the JS file passes lint.

Commit:
fix(report): balance report UI fixes

This commit fixes some broken translation on the balance report modal as
well as invalid HTML.

sfount committed:
chore(journal) Refactor translation;rm legacy code

This commit updates all hardcoded or incorrectly entered translations
for the new journal features.

lomamech committed:
Update html of price list to keep all functionality in Small Screen

lomamech committed:
Update function of translation calling

sfount committed:
Merge remote-tracking branch 'ima/master' into feature-journal-editing

Conflicts:
	client/src/partials/journal/journal.html
	client/src/partials/journal/journal.js

Commit:
refactor(journal): disable header on edit

This commit improves the safety of the user experience by disabling all
of the top bar while editing.  This prevents the user from accidentally
ungrouping the transactions, performing an inline filter that messes
with their edits.

Commit:
fix(journal): disable cellNav on custom columns

This commit disables cellNavigation on the plugin columns by getting
them by name and manually switching them off.  It uses a $timeout() hack
to try and guarantee this takes effect before the user is ready to click
on the page.

Commit:
fix(report): accounts report gives the document

This commit finally fixes the cumulative SUM in the accounts report as
well as putting in a new column with the originating document number.
This will allow administrators to really quickly look up the details of
that transaction.

jniles committed:
Merge pull request #1262 from IMA-WorldHealth/feature-journal-editing

(feature) Posting Journal editing

Commit:
fix(build): build in development mode

This commit fixes a bug with the gulp build that does not register the
correct build environment for setting the angular debug flags.
Previously, it was always true.

Commit:
fix(cashflow): do not filter reversals

This commit fixes the cashflow report by not filtering by reversals.  If
the reversal query is NULL, it will simply print it inline.  However,
previously the NULL killed the entire report.

This should be investigated... but at least our clients can use the
report.

jniles committed:
Merge pull request #1264 from jniles/fix-build-without-debug-mode

fix(build): build in development mode

lomamech committed:
Sanitaza code
Using the common class for button
using font awesome instead glyphicons
Using not necessary colspan

Commit:
fix(accounts): fix report date handling

This commit fixes the account report's date handling so that it limits
to the correct dates.  It is a critical bug fix for Vanga.

Caused by a variation on #1274.

Commit:
fix(accounts): nicer report UI

This commit cleans up the report UI by removing icons and making the
checkoxes uniform with other modules that use checkboxes.

Commit:
fix(income/expense): allow any cashbox in report

This commit removes the restriction on only primary cashboxes.  A report
on auxiliary cashboxes may yield little data, but this limitation
should not be built into the software.

sfount committed:
fix(build) call asset hash/ movement on JS watch

This commit replaces direct Javascript compilation with re-versioning
and compiling assets on every javascript file change. This ensures that
the changes are factored into the build being used by the current
server.

Commit:
refactor(html): use state not href

This commit refactors a ton of client modules to use `ui-sref` tags
which navigate to particular states instead of `href`s which are
dependent on our URL syntax.

Additionally, it deletes two unused files (profit centers and an
accounts template).

It also updates several HTML files to use the translate directive
instead of the translate filter.

Closes #1279.  Closes #1223.

Commit:
refactor(report): Make balance dates prominent

This commit refactors the balance report so that the date ranges are
more prominent.  It also cleans up the code a bit by using let/const
instead of var.

Ideally, this should all by migrated to SQL.  There is also some
confusing behavior - see #1293.

Partially addresses #1199.

Commit:
refactor(reports): account report date prominence

This commit makes the account report's date range a bit more prominent.
It should help on lower-end printers.

Partially addresses #1199.

Commit:
refactor(reports): Aged Debtor date made prominent

This commit makes the aged debtor date more prominent on the report.

Partially address #1199.

DedrickEnc committed:
amending the PR by using filterParser

Commit:
feat(vouchers): implement voucher search modal

This commit implements the voucher search modal.  The following fields
are searchable:
 1. Date (range)
 2. Reference
 3. Description
 4. User

Closes #671.

Commit:
refactor(search): implement bhClear link/button

This commit implements `<bh-clear on-clear="callback()"></bh-clear>`.
This little component replaces clear links on the search modals and
reduces the amount of code developers have to write.

Commit:
test(vouchers): implement search end-to-end tests

This commit implements a few end-to-end tests for the voucher search
registry.  These tests are relatively few and should be expanded with
further functionality added in the future.

Commit:
chore(docs): removed CLAs

This commit removes the CLAs to reduce confusion when trying to
contribute to the repository.  We hope to introduce some higher
contributing standards as we get more experience in the area.

Commit:
fix(reports): account report running balance

This commit implements the correct cumulative balance summation for the
accounts report that initializes and updates variables in the same SQL
statement. By maintaining all variables in the same statement, the
variables are ensured to be zeroes out prior to usage and do not spread
over multiple MySQL connections.

This solution, while complex, allows total control over filtering via
our standard filtering solutions.  Placing the query in a Stored
Procedure (another solution to localize the variable to a single MySQL
connection/context) would require that any further filters write a new
interface to the stored procedure.

Closes #1273.

sfount committed:
fix(translation) update patient group assignment

This commit updates the french translation for editing a patient's group
assignments (patients and debtor groups). It also adds placeholder text
for patient group searching.

sfount committed:
fix(validation) registration location error state

This commit ensures the error state on patient registration location
select is shown. Previously a sub form that never has its state
'submitted' set was used for the validation trigger. The parent form
should be used as this is the one that angular processes.

DedrickEnc committed:
fixing integration test

Commit:
feat(patients): add cumsum to patient activity

This commit adds a cumulative summation to the Patient Activity page. It
should help the auditors track what the patient's balance is a lot
easier and at what point they had paid their bill in full.

Closes #1248.

jniles committed:
docs: Update CONTRIBUTING

This commit removes old email addresses and so forth to aid first time contributors.
